src/main/java/org/ow2/proactive/catalog/report/CatalogObjectReportPDFGenerator.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.catalog.report;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.TreeSet;
+
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.apache.pdfbox.pdmodel.common.PDRectangle;
+import org.ow2.proactive.catalog.dto.CatalogObjectMetadata;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import be.quodlibet.boxable.BaseTable;
+import be.quodlibet.boxable.datatable.DataTable;
+
+
+@Component
+public class CatalogObjectReportPDFGenerator {
+
+    private static final List<String> EMPTY_ROW = Arrays.asList("", "", "", "", "");
+
+    private static final List<String> HEADER = Arrays.asList("Bucket Name",
+                                                             "Project Name",
+                                                             "Object name",
+                                                             "Object kind",
+                                                             "Content Type");
+
+    private static final PDRectangle PAGE_SHAPE_A4 = new PDRectangle(PDRectangle.A4.getHeight(),
+                                                                     PDRectangle.A4.getWidth());
+
+    public byte[] generatePDF(TreeSet<CatalogObjectMetadata> orderedObjectsPerBucket) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The type of the "orderedObjectsPerBucket" object should be an interface such as "SortedSet" rather than the implementation "TreeSet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)